src/jailer/src/env.rs

```diff
@@ -59,6 +59,27 @@ fn dup2(old_fd: libc::c_int, new_fd: libc::c_int) -> Result<()> {
         .map_err(Error::Dup2)
 }
 
+// This is a wrapper for the clone system call. When we want to create a new process in a new
+// pid namespace, we will call clone with a NULL stack pointer. We can do this because we will
+// not use the CLONE_VM flag, this will result with the original stack replicated, in a similar
+// manner to the fork syscall. The libc wrapper prevents use of a NULL stack pointer, so we will
+// call the syscall directly.
+fn clone(child_stack: *mut libc::c_void, flags: libc::c_int) -> Result<libc::c_int> {
+    // Clone parameters order is different between x86_64 and aarch64.
+    #[cfg(target_arch = "x86_64")]
+    return SyscallReturnCode(unsafe {
+        libc::syscall(libc::SYS_clone, flags, child_stack, 0, 0, 0) as libc::c_int
+    })
+    .into_result()
+    .map_err(Error::Clone);
+    #[cfg(target_arch = "aarch64")]
+    return SyscallReturnCode(unsafe {
+        libc::syscall(libc::SYS_clone, flags, child_stack, 0, 0, 0) as libc::c_int
+    })
+    .into_result()
+    .map_err(Error::Clone);
```

Comment:
@acatangiu Are you sure we want to unify the specific arch wrapper ?
I think its a bad habit, it will be super easy to insert a bug by changing one of the pid/ctid/new tls args.
I believe we should leave it that way, to prevent future bugs.
If you still insist, I will do it, I want to make that commit happen finally :)
@serban300 - can you approve the changes too in order to move on ? thanks.

Reply:
Personally I agree that it would be safer to use different logic for `x86_64` vs `aarch64`. Also the PR looks good.